### ISSUE TITLE
Create automated process in the open to refresh extensions repo with openjdk changes

### ISSUE THREAD

**Issue by archanaNogriya** [ibmruntimes/openj9-openjdk-jdk8#10]
Create job configuration on Eclipse openj9 CI (ci.eclipse.org/openj9) Jenkins to run "mirror" script - https://github.ibm.com/open-class-libraries/openjdk-build/blob/master/git-hg/update8.sh

(Note : this is bash script)

It requires the jenkins job to provide 2 variables

$GITHUB_IBM (eg https://github.ibm.com/open-class-libraries)
$J9REPOSITORY (eg openJDK8u152-b16-update-test)

**Comment by archanaNogriya** [comment]
Something similar we tried in our local jenkin: https://hyc-rt-openjdk-jenkins.swg-devops.com/job/Update_OpenJ9/configure

**Comment by archanaNogriya** [comment]
@AdamBrousseau @DanHeidinga  issues is now created here

**Comment by archanaNogriya** [comment]
@AdamBrousseau , once we have this Job pipeline in place then before running directly into actual source code we can clone that repo locally and then run this job to make sure it is doing the right thing :). however issue would be I dont have access to run the job in Jenkin. So might need someone help to test until if we have option and i can do that.

**Comment by AdamBrousseau** [comment]
@archanaNogriya that script is in an internal repository that I do not have access to. Please push it to the openjdk8 repo (at least on your fork for now) so we can look and point at it.

**Comment by archanaNogriya** [comment]
Ahh @AdamBrousseau let me do that.. sorry :) should have done that before

**Comment by archanaNogriya** [comment]
@AdamBrousseau you can find the script here now https://github.com/ibmruntimes/openj9-openjdk-jdk8/tree/openj9/scripts

**Comment by archanaNogriya** [comment]
@AdamBrousseau , is there any updates in this? To update you regarding few action on this script, there needs to be changes made with regard to how it handles the tagging. Basically it needs to reset the head to the new tag level before doing the merge. Otherwise we could get ourselves into the position where our tagged levels are not the same as Oracle's tagged levels.

I have made changes what I think is required and should work. But I need to test my changes. 
So incase if you already working on this, you can continue. once I have my updates script tested I will let you know and you can update the script with latest one. if that works. 

Attaching the updated script so you can see what I have made change based on what I explained on tagging. Remember this attached script is not tested yet so not sure the changes I have made will work :)

@andrew-m-leonard 

[update8.zip](https://github.com/ibmruntimes/openj9-openjdk-jdk8/files/1600206/update8.zip)

**Comment by archanaNogriya** [comment]
- Guidance for running mirror script
 -- Install Mercurial:
```
sudo apt-get update
sudo apt-get install mercurial
Create a directory ~/hg_ext for Mercurial extensions
cd ~/hg_ext and download:
http://hg.openjdk.java.net/code-tools/jcheck/dist/raw-file/tip/jcheck.py
http://hg.openjdk.java.net/code-tools/trees/raw-file/tip/trees.py
Edit/create file ~/.hgrc with contents:
[ui]
username = <yourusername>

[extensions]
hgext.color=
hgext.graphlog=
hgext.mq=
hgext.purge=
mq = 
jcheck = ~/hg_ext/jcheck.py
trees = ~/hg_ext/trees.py

[alias]
slog = log --template '[{rev}] {node|short} {branch} ({tags}) {desc|firstline}\n'
sout = outgoing --template '[{rev}] {node|short} {branch} ({tags}) {desc|firstline}\n'
sin = incoming --template '[{rev}] {node|short} {branch} ({tags}) {desc|firstline}\n'
rva = revert --all --no-backup
# tell the based oracle level
base($1) =log -r 'parents(ancestors("$1") and keyword("ibm.start"))'
# tell the first changeset in an ibm spur
first($1)=log -r 'ancestors("$1") and keyword("ibm.start")'
```
-- Obtain git-remote-hg Mercurial plugin
```
wget -O $HOME/bin/git-remote-hg https://raw.github.com/felipec/git-remote-hg/master/git-remote-hg
chmod +x $HOME/bin/git-remote-hg
export PATH=$HOME/bin:$PATH
```
-- Getting latest git
Note- since merge option --allow-unrelated-histories does not work in below git2.9 hence you need 2.9 or above version

```
$ sudo add-apt-repository ppa:git-core/ppa
$ sudo apt-get update
$ sudo apt-get install git
```
Latest Script file with Tag code included 
[update8.zip](https://github.com/ibmruntimes/openj9-openjdk-jdk8/files/1611552/update8.zip)

**Comment by AdamBrousseau** [comment]
This is a bit of a roadblock. The machine I was planning on using to run this script was the master of our Jenkins instance. We do not own this machine and do not have root access on it. It currently has git version 1.9.0 and no Mercurial installed. We could open an Eclipse ticket to see if they will install hg and update git for us but no guarantees. We may also be able to install them to $HOME similar to the git-remote-hg story above.

We could run this on our pLinux or zLinuz machines but I would rather reserve those machine for real builds.

**Comment by AdamBrousseau** [comment]
There is also a problem with your change to the script. You are referencing `$NEWTAG` before it is defined. Please open a pull request with your update so it is easier to discuss the changes.

**Comment by archanaNogriya** [comment]
Sure @AdamBrousseau  , let me open the pull request.

**Comment by AdamBrousseau** [comment]
I managed to get the script working on my laptop. Need git > 2.9, hg and git-remote-hg. Took 1h40m to run.

@DanHeidinga what are your thoughts on where to run this?
My preference is master but we would need Eclipse to install newer git and hg on the master. Either that or we can look at installing to $HOME without root, which is what the instructions currently do for git-remote-hg.
Other option is to run on one/all of the p/z machines.

I imagine we would schedule this overnight so it wouldn't interfere too much with PRs if we did decide to run it on our machines.

**Comment by AdamBrousseau** [comment]
>where to run this?

I have created a [proxy_worker](https://ci.eclipse.org/openj9/computer/proxy_worker/) machine for another task out of our P pool we have access to. We can use this machine to run this script.

**Comment by archanaNogriya** [comment]
@AdamBrousseau I will let you know once we have merged openj9-openjdk-jd8 repo open, Due to merge conflict we had to do this manual.  

What I think best we can do after this, taking latest merged one as base and do the test run using your jenkin job and see how that works.  Let me know what you think?

Also After getting this merge issue the question raise about error notification,  Is there way if in future we get this kind of conflict issue or any other merge problems, if this can be notify in slack channel or so? I know we have way to do jenkins job and slack notification, but not sure how we do that. 
May be something like this Notification Plugins ? https://wiki.jenkins.io/pages/viewpage.action?pageId=73203737
 

I think that is something we can integrate with our jobs?
